participant/app/cookies.server.ts

```diff
@@ -0,0 +1,78 @@
+// app/sessions.js
+import { createCookie } from "@remix-run/node"; // or "@remix-run/cloudflare"
+import { redirect } from "react-router";
+import { v4 as uuidv4 } from "uuid";
+import { findSubmission, upsertSubmission } from "./utils/db.server";
+const MAX_SESSION_SECONDS = Number(process.env.MAX_SESSION_SECONDS) || 1800;
+
+type ParticipantCookieContents = {
+  submissionID?: string;
+};
+
+export const ParticipantCookie = createCookie("prp-recertification-form", {
+  secure: true,
+});
+
+export const sessionCheck = (time: Date): boolean => {
+  const age = (new Date().getTime() - time.getTime()) / 1000;
+  if (age > MAX_SESSION_SECONDS) {
+    console.log(
+      `Session is not so fresh 🤢: ${age} seconds, max ${MAX_SESSION_SECONDS}`
+    );
+    return false;
+  }
+  return true;
+};
+
+export const cookieParser = async (request: Request, resetSession = false) => {
+  const cookie = ((await ParticipantCookie.parse(
+    request.headers.get("Cookie")
+  )) || {}) as ParticipantCookieContents;
+  let forceRedirect: boolean = resetSession;
+  const urlId = "gallatin"; // ONLY HERE until PRP-227
+  if (cookie) {
+    if (cookie.submissionID) {
```

Comment:
I think you don't have a unit test for:
- if there is no cookie (if `cookie == {}`)
- if there is a cookie but it has no submission ID (if `cookie.submissionID == undefined`)